Source/DataSources/RectangleGeometryUpdater.js

```diff
@@ -203,6 +232,12 @@ define([
         options.granularity = defined(granularity) ? granularity.getValue(Iso8601.MINIMUM_VALUE) : undefined;
         options.stRotation = defined(stRotation) ? stRotation.getValue(Iso8601.MINIMUM_VALUE) : undefined;
         options.rotation = defined(rotation) ? rotation.getValue(Iso8601.MINIMUM_VALUE) : undefined;
+        options.offsetAttribute = GeometryHeightProperty.computeGeometryOffsetAttribute(height, extrudedHeight, Iso8601.MINIMUM_VALUE);
+
+        if (extrudedHeight instanceof GeometryHeightProperty && Property.getValueOrDefault(extrudedHeight.height, Iso8601.MINIMUM_VALUE, HeightReference.NONE) === HeightReference.CLAMP_TO_GROUND) {
+            scratchRectangleGeometry.setOptions(options);
+            options.extrudedHeight = GeometryHeightProperty.getMinimumTerrainValue(scratchRectangleGeometry.rectangle);
```

Comment:
This is confusing, why are we setting extrudedHeight after we pass it to setOptions?  Does order matter here because it looks like a bug if not and is confusing if so.

Reply:
Obviously this and other comments apply to the other geometries as well, but I won't comment anywhere except RectangleGeoemtryUpdater to avoid noise.

Reply:
OK, I see what's going on here, but this seems like a really backwards way to do this.  We are using a scratch geometry instance simply so we can call `.rectangle` on it to pass to `GeometryHeightProperty.getMinimumTerrainValue`?  This makes me think that something is being done at the wrong level of abstraction.  Why can't we add an option to the geometries so that we don't have to create a scratch one just for this use case?

Reply:
> Why can't we add an option to the geometries so that we don't have to create a scratch one just for this use case

What do you mean by this?  I don't get it.

Reply:
If I'm reading the code correctly, we are using a `scratchRectangleGeometry` to compute the `rectangle` property which we then use to compute another value and set it back onto options of a different RectangleGeometry that gets created later.  That seems a little behind your back to get to your elbow to me.

It also means that rectangle gets computed up front for the geometry on the main thread (and I assume computed again later during tessellation?)  Why doesn't the geometry just take the options needed to do the extrudedHeight computation itself during tessellation?

There could be a good reason we're doing it this way, but from the outside looking in it seems like there needs to be a cleaner way.

Reply:
Addressed in #6733